modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ObjectFactoryClassRepository.java

```diff
@@ -90,42 +90,52 @@
      * defined. This way our API does not expose spring specific classes.
      */
-    Enhancer enhancer = new Enhancer();
-    // Use SmartFactoryBean since it's the only way to force spring to pre-instantiate FactoryBean for singletons
-    enhancer.setInterfaces(new Class[] {SmartFactoryBean.class});
-    enhancer.setSuperclass(objectFactoryType);
-    enhancer.setCallbackType(MethodInterceptor.class);
-    if (SmartFactoryBean.class.getClassLoader() != objectFactoryType.getClassLoader()) {
-      // CGLIB needs access to both the spring interface and the extended factory class.
-      // If the factory class is defined in a plugin, its classloader has to be passed.
-      enhancer.setClassLoader(new CompositeClassLoader(ObjectFactoryClassRepository.class.getClassLoader(),
-                                                       objectFactoryType.getClassLoader()));
-    }
-    enhancer.setUseCache(false);
-    Class<ObjectFactory> factoryBeanClass = enhancer.createClass();
-    registerStaticCallbacks(factoryBeanClass, new Callback[] {
-        (MethodInterceptor) (obj, method, args, proxy) -> {
-          if (method.getName().equals("isSingleton")) {
-            return !componentBuildingDefinition.isPrototype();
-          }
-          if (method.getName().equals("getObjectType") && !ObjectTypeProvider.class.isAssignableFrom(obj.getClass())) {
-            return createdObjectType;
-          }
-          if (method.getName().equals("getObject")) {
-            Object createdInstance = proxy.invokeSuper(obj, args);
-            instancePostCreationFunction.accept(createdInstance);
-            return createdInstance;
-          }
-          if (method.getName().equals("isPrototype")) {
-            return componentBuildingDefinition.isPrototype();
-          }
-          if (method.getName().equals("isEagerInit")) {
-            return !isLazyInitFunction.get();
+    ClassLoader originalClassLoader = null;
+    try {
+      Enhancer enhancer = new Enhancer();
+      // Use SmartFactoryBean since it's the only way to force spring to pre-instantiate FactoryBean for singletons
+      enhancer.setInterfaces(new Class[] {SmartFactoryBean.class});
+      enhancer.setSuperclass(objectFactoryType);
+      enhancer.setCallbackType(MethodInterceptor.class);
+      if (SmartFactoryBean.class.getClassLoader() != objectFactoryType.getClassLoader()) {
+        // CGLIB needs access to both the spring interface and the extended factory class.
+        // If the factory class is defined in a plugin, its classloader has to be passed.
+        CompositeClassLoader loader = new CompositeClassLoader(ObjectFactoryClassRepository.class.getClassLoader(),
+                                                               objectFactoryType.getClassLoader());
+        enhancer.setClassLoader(loader);
+        originalClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
This must be done above in the variable declaration and not here. If not in the finally clause you may be setting a null.

Reply:
but I'm checking that originalClassLoader is not null in the finally block.